### PR TITLE
otelhttp: use plain fields

### PR DESCRIPTION
**Pull request by ash2k**
Fixes https://github.com/open-telemetry/opentelemetry-go-contrib/issues/4502.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -12,6 +12,10 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 
 - Add the new `go.opentelemetry.io/contrib/instrgen` package to provide auto-generated source code instrumentation. (#3068, #3108)
 
+### Changed
+
+- `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` uses plain fields for measures (rather than maps). (#4615)
+
```
Comment (pellared) [inline CHANGELOG.md:18]: Changelog entries for internal changes are not needed.

File: instrumentation/net/http/otelhttp/handler.go [hunk 2]
```diff
@@ -35,18 +35,19 @@ type middleware struct {
...
+	spanStartOptions     []trace.SpanStartOption
+	readEvent            bool
+	writeEvent           bool
+	filters              []Filter
+	spanNameFormatter    func(string, *http.Request) string
+	requestBytesCounter  metric.Int64Counter
+	responseBytesCounter metric.Int64Counter
+	serverLatencyMeasure metric.Float64Histogram
+	publicEndpoint       bool
+	publicEndpointFn     func(*http.Request) bool
```
Comment (pellared) [inline instrumentation/net/http/otelhttp/handler.go:50]: nit: I propose to make a separate "section" for the instruments


```suggestion
	tracer            trace.Tracer
	meter             metric.Meter
	propagators       propagation.TextMapPropagator
	spanStartOptions  []trace.SpanStartOption
	readEvent         bool
	writeEvent        bool
	filters           []Filter
	spanNameFormatter func(string, *http.Request) string
	publicEndpoint    bool
	publicEndpointFn  func(*http.Request) bool
	
	requestBytesCounter  metric.Int64Counter
	responseBytesCounter metric.Int64Counter
	serverLatencyMeasure metric.Float64Histogram
```

**Review by pellared — APPROVED** [review]
Thanks 👍

**Review by pellared — APPROVED** [review]
👍

**Review by hanyuancheung — APPROVED** [review]

**Pull request merged**